packages/components/src/components/ViewSettings.tsx

```diff
@@ -5,49 +5,137 @@ import { InputItem } from "./ui/InputItem";
 import { Checkbox } from "./ui/Checkbox";
 import { HeadedSection } from "./ui/HeadedSection";
 import { Text } from "./ui/Text";
+import { MergedItemSettings, LocalItemSettings } from "./SquiggleViewer/utils";
 import { defaultTickFormat } from "../lib/distributionSpecBuilder";
 
 export const viewSettingsSchema = yup.object({}).shape({
   chartHeight: yup.number().required().positive().integer().default(350),
-  showSummary: yup.boolean().required(),
-  showEditor: yup.boolean().required(),
-  logX: yup.boolean().required(),
-  expY: yup.boolean().required(),
-  tickFormat: yup.string().default(defaultTickFormat),
+  showSummary: yup.boolean().required().default(false),
+  logX: yup.boolean().required().default(false),
+  expY: yup.boolean().required().default(false),
+  tickFormat: yup.string().required().default(defaultTickFormat),
   title: yup.string(),
   minX: yup.number(),
   maxX: yup.number(),
+  xAxisType: yup
+    .mixed<"number" | "dateTime">()
+    .oneOf(["number", "dateTime"])
+    .default("number"),
   distributionChartActions: yup.boolean(),
   diagramStart: yup.number().required().positive().integer().default(0).min(0),
   diagramStop: yup.number().required().positive().integer().default(10).min(0),
   diagramCount: yup.number().required().positive().integer().default(20).min(2),
 });
 
-type FormFields = yup.InferType<typeof viewSettingsSchema>;
+export type EditableViewSettings = yup.InferType<typeof viewSettingsSchema>;
+
+export const viewSettingsToMerged = (
+  settings: EditableViewSettings
```

Comment:
I'm not sure I'm such a fan of refactoring all the props into a hierarchical MergedItemSettings. If MergedItemSettings simply matched the long flat record of the props then we could delete all this conversation code that goes between the two. I think Salva wanted to refactor props into a hierarchy, but I don't see why that would be beneficial? We can restrict with fields a function can use with types anyway.

Reply:
Yeah, I'd prefer to go the opposite way.

Definitely agree that the shape of settings and the shape of props should match and this conversion shouldn't be necessary.

But I'd prefer the props to become hierarchical, for the following reasons:
- easier to trace which part configures what; `diagramStart` is not really function-specific
- shorter property names; it's weird for `SquiggleSpecificSubcomponent` to have properties named `specificSubcomponentProp`, and that's what we'll end up with if we continue with flat objects; easier to pass the specific object, `<SquiggleSpecificSubcomponent {...props.specificSubcomponentSettings} />`
- as I mentioned another comment, it's weird to pass the entire object to subcomponents and functions that care only about a few fields from those; there's no big problem with it, but it'll be visible e.g. in React dev tools and in debugging `console.log` and so on
  - it's also incompatible with using `React.memo`, right now it's not a problem but might make some optimizations harder in the future
- Typescript's type intersections are great but they make code less readable; e.g.:

<img width="954" alt="Screen Shot 2022-10-18 at 22 05 21" src="https://user-images.githubusercontent.com/89368/196509994-3b866b4d-b600-4ec9-8759-f52ea49cf3e6.png">